components/camera/camera.go

```diff
@@ -438,16 +436,13 @@ func (c *reconfigurableCamera) DoCommand(ctx context.Context, cmd map[string]int
 }
 
 // Reconfigure reconfigures the resource.
-func (c *reconfigurableCamera) Reconfigure(ctx context.Context, newCamera resource.Reconfigurable) error {
+func (c *reconfigurableCamera) Reconfigure(_ context.Context, newCamera resource.Reconfigurable) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	actual, ok := newCamera.(*reconfigurableCamera)
 	if !ok {
 		return utils.NewUnexpectedTypeError(c, newCamera)
 	}
-	if err := viamutils.TryClose(ctx, c.actual); err != nil {
-		golog.Global().Errorw("error closing old", "error", err)
-	}
```

Comment:
This was another tricky bug to find. Closing the old camera _after_ opening the new camera will also close the new camera. Because this function takes in a new, already-opened camera there's no way to close the camera by the time you've reached this line without destroying the new one. I leave closing the old camera to the caller.

The reason this happens is because both the old and new camera point to the same path (or file) on Linux. If the old camera closes that file, then the new camera will get an EOF error when it tires to read from it. Opening a camera for streaming is non-trivial as well. Ultimately, we'd have call [Open](https://github.com/pion/mediadevices/blob/5b995002908809bb17f82e88fcfab8b9d4ca33a9/pkg/driver/camera/camera_linux.go#L151) and [StartStreaming](https://github.com/pion/mediadevices/blob/5b995002908809bb17f82e88fcfab8b9d4ca33a9/pkg/driver/camera/camera_linux.go#L151) which we already do when we create a new driver. Because `StartStreaming` will need to know which driver properties to start streaming with, we'd have to go through the entire discovery process _again_ if we want to stream the right image properties from the driver. It seemed simpler to just leave closing the camera to the caller, call close on the old camera _then_ open the new camera.

Reply:
I don't think we want an Open method. We expect that to be a construction time thing and it seems overly specific to webcams right now.

Where is Close happening now? You mentioned the caller but I don't see where it moved to. Seems like we'd have a FD leak now.

Reply:
> I don't think we want an Open method. We expect that to be a construction time thing and it seems overly specific to webcams right now.

Gotcha. Makes sense.

> Where is Close happening now? You mentioned the caller but I don't see where it moved to. Seems like we'd have a FD leak now.

This happens on line 313. I moved it inside of `reconfigureCamera` so it's easier to spot.